packages/components/src/combobox-control/test/index.js

```diff
@@ -0,0 +1,141 @@
+	{ name: 'Middle East Time', abbreviation: 'MET' },
+	{ name: 'Near East Time', abbreviation: 'NET' },
+	{ name: 'Pakistan Lahore Time', abbreviation: 'PLT' },
+	{ name: 'India Standard Time', abbreviation: 'IST' },
+	{ name: 'Bangladesh Standard Time', abbreviation: 'BST' },
+	{ name: 'Vietnam Standard Time', abbreviation: 'VST' },
+	{ name: 'China Taiwan Time', abbreviation: 'CTT' },
+	{ name: 'Japan Standard Time', abbreviation: 'JST' },
+	{ name: 'Australia Central Time', abbreviation: 'ACT' },
+	{ name: 'Australia Eastern Time', abbreviation: 'AET' },
+	{ name: 'Solomon Standard Time', abbreviation: 'SST' },
+	{ name: 'New Zealand Standard Time', abbreviation: 'NST' },
+	{ name: 'Midway Islands Time', abbreviation: 'MIT' },
+	{ name: 'Hawaii Standard Time', abbreviation: 'HST' },
+	{ name: 'Alaska Standard Time', abbreviation: 'AST' },
+	{ name: 'Pacific Standard Time', abbreviation: 'PST' },
+	{ name: 'Phoenix Standard Time', abbreviation: 'PNT' },
+	{ name: 'Mountain Standard Time', abbreviation: 'MST' },
+	{ name: 'Central Standard Time', abbreviation: 'CST' },
+	{ name: 'Eastern Standard Time', abbreviation: 'EST' },
+	{ name: 'Indiana Eastern Standard Time', abbreviation: 'IET' },
+	{ name: 'Puerto Rico and US Virgin Islands Time', abbreviation: 'PRT' },
+	{ name: 'Canada Newfoundland Time', abbreviation: 'CNT' },
+	{ name: 'Argentina Standard Time', abbreviation: 'AGT' },
+	{ name: 'Brazil Eastern Time', abbreviation: 'BET' },
+	{ name: 'Central African Time', abbreviation: 'CAT' },
+];
+const mapTimezoneOption = ( tz ) => ( {
+	value: tz.abbreviation,
+	label: tz.name,
+} );
+
+const timezoneOptions = timezones.map( mapTimezoneOption );
+const defaultLabelText = 'Select a timezone';
+const getLabel = () => screen.getByText( defaultLabelText );
+const getInput = () => screen.getByRole( 'combobox' );
+const getRenderedOptions = () => screen.getAllByRole( 'option' );
+
+describe( 'ComboboxControl', () => {
+	// Use real timers to avoid timeout errors from userEvent
+	beforeEach( () => {
+		jest.useRealTimers();
+	} );
+
+	const TestComboboxControl = ( props ) => {
+		const [ value, setValue ] = useState( null );
+		return (
+			<>
+				<ComboboxControl
+					{ ...props }
+					value={ value }
+					label={ defaultLabelText }
+					options={ timezoneOptions }
+					onChange={ setValue }
+				/>
+				<p data-testid="value-output">{ value }</p>
+			</>
+		);
+	};
+
+	it( 'should render', () => {
+		render( <TestComboboxControl /> );
+
+		const input = getInput();
+		expect( input ).toBeVisible();
+	} );
+
+	it( 'should render with visible label', () => {
+		render( <TestComboboxControl /> );
+		const label = getLabel();
+		expect( label ).toBeVisible();
+	} );
+
+	it( 'should render with hidden label', () => {
+		render( <TestComboboxControl hideLabelFromVision={ true } /> );
+		const label = getLabel();
+
+		expect( label ).toBeInTheDocument();
+		expect( label ).toHaveAttribute(
+			'data-wp-component',
+			'VisuallyHidden'
+		);
```

Comment:
I did! Unfortunately, `not.toBeVisible()` doesn't work here, because we're not doing [any of the things that satisfy that assertion](https://github.com/testing-library/jest-dom#tobevisible). Instead, our method of hiding the label relies on some clipping and absolute positioning, but as far as the test is concerned it's still technically visible.

Is that something we should look into addressing in a followup, so we can use a less implementation-focused test?

Reply:
I guess so. Given that we have [a snapshot test](https://github.com/WordPress/gutenberg/blob/d5de22f316c1049f46675b52e1436890a16d7a5d/packages/components/src/visually-hidden/test/index.tsx) for `VIsuallyHidden` printing out the exact styles, I guess we can leave the code in this test as-is for the time being